src/http/registration.js

```diff
@@ -2,37 +2,21 @@ CoSeMe.namespace('registration', (function(){
   'use strict';
 
   function getToken(phone) {
-    var signature = atob(
-      'MIIDMjCCAvCgAwIBAgIETCU2pDALBgcqhkjOOAQDBQAwfDELMAkGA1UEBhMCVVMxEzARBgNVBAgTCkNhbGlmb3JuaWExFDASBgNVBAcTC1NhbnRhIENsYXJhMRYwFAYDVQQKEw1XaGF0c0FwcCBJbmMuMRQwEgYDVQQLEwtFbmdpbmVlcmluZzEUMBIGA1UEAxMLQnJpYW4gQWN0b24wHhcNMTAwNjI1MjMwNzE2WhcNNDQwMjE1MjMwNzE2WjB8MQswCQYDVQQGEwJVUzETMBEGA1UECBMKQ2FsaWZvcm5pYTEUMBIGA1UEBxMLU2FudGEgQ2xhcmExFjAUBgNVBAoTDVdoYXRzQXBwIEluYy4xFDASBgNVBAsTC0VuZ2luZWVyaW5nMRQwEgYDVQQDEwtCcmlhbiBBY3RvbjCCAbgwggEsBgcqhkjOOAQBMIIBHwKBgQD9f1OBHXUSKVLfSpwu7OTn9hG3UjzvRADDHj+AtlEmaUVdQCJR+1k9jVj6v8X1ujD2y5tVbNeBO4AdNG/yZmC3a5lQpaSfn+gEexAiwk+7qdf+t8Yb+DtX58aophUPBPuD9tPFHsMCNVQTWhaRMvZ1864rYdcq7/IiAxmd0UgBxwIVAJdgUI8VIwvMspK5gqLrhAvwWBz1AoGBAPfhoIXWmz3ey7yrXDa4V7l5lK+7+jrqgvlXTAs9B4JnUVlXjrrUWU/mcQcQgYC0SRZxI+hMKBYTt88JMozIpuE8FnqLVHyNKOCjrh4rs6Z1kW6jfwv6ITVi8ftiegEkO8yk8b6oUZCJqIPf4VrlnwaSi2ZegHtVJWQBTDv+z0kqA4GFAAKBgQDRGYtLgWh7zyRtQainJfCpiaUbzjJuhMgo4fVWZIvXHaSHBU1t5w//S0lDK2hiqkj8KpMWGywVov9eZxZy37V26dEqr/c2m5qZ0E+ynSu7sqUD7kGx/zeIcGT0H+KAVgkGNQCo5Uc0koLRWYHNtYoIvt5R3X6YZylbPftF/8ayWTALBgcqhkjOOAQDBQADLwAwLAIUAKYCp0d6z4QQdyN74JDfQ2WCyi8CFDUM4CaNB+ceVXdKtOrNTQcc0e+t'
-    );
-    var classesMd5 = atob('U8Rv0Yqm6qUsIGkGbBBaZA==');
-    var key2 = atob('/UIGKU1FVQa+ATM2A0za7G2KI9S/CwPYjgAbc67v7ep42eO/WeTLx1lb1cHwxpsEgF4+PmYpLd2YpGUdX/A2JQitsHzDwgcdBpUf7psX1BU=');
-    var data = CryptoJS.enc.Latin1.parse(signature + classesMd5 + phone);
-
-    var opad = new Uint8Array(64);
-    var ipad = new Uint8Array(64);
-    for (var i = 0; i < 64; i++) {
-      opad[i] = 0x5C ^ key2.charCodeAt(i);
-      ipad[i] = 0x36 ^ key2.charCodeAt(i);
-    }
-    ipad = CryptoJS.enc.UInt8Array.parse(ipad);
-    opad = CryptoJS.enc.UInt8Array.parse(opad);
-
-    var output = CryptoJS.SHA1(
-      opad.concat(CryptoJS.SHA1(ipad.concat(data)))
-    );
-
-    return output.toString(CryptoJS.enc.Base64);
+    return CryptoJS.MD5("PdA2DJyKoUrwLw1Bg6EIhzh502dF9noR9uFCllGk1418865329241"+phone).toString();
   }
 
+
   function getRealDeviceId(aSeed) {
     var seed = aSeed || (Math.random() * 1e16).toString(36).substring(2, 10);
-    var id = CryptoJS.SHA1(seed).toString(CryptoJS.enc.Latin1).substring(0, 20);
+    var id = CryptoJS.SHA1(seed).toString(CryptoJS.enc.Latin1).substring(0, 20)
+      .split('').map(function (e) {
+        return String.fromCharCode(e.charCodeAt(0) % 128)
```

Comment:
Although is here where is the real problem: CoSeMe is sending more id bytes than needed, I fear by manipulating the SHA1 we were loosing the cryptographic properties of the hash. We should manipulate the way in which this parameter is being encoded in the URL to avoid unnecessary expansions made by [`encodeURIComponent`](https://github.com/mozillahispano/coseme/blob/master/src/common/utils.js#L327).